Tasks/AzureRmWebAppDeployment/MSDeployUtility.ts

```diff
@@ -0,0 +1,139 @@
+                             removeAdditionalFilesFlag: boolean, excludeFilesFromAppDataFlag: boolean, takeAppOfflineFlag: boolean,
+                             virtualApplication: string, setParametersFile: string, additionalArguments: string) : string {
+
+    var msDeployCmdArgs = ' -verb:sync';
+    msDeployCmdArgs += ' -source:package=\'' + packageFile + '\'';
+    msDeployCmdArgs += ' -dest:auto,ComputerName=https://' + azureRMWebAppConnectionDetails["KuduHostName"] + '/msdeploy.axd?site=' + webAppNameForMSDeployCmd + ',';
+    msDeployCmdArgs += 'UserName=' + azureRMWebAppConnectionDetails['UserName'] + ',Password=' + azureRMWebAppConnectionDetails['UserPassword'] + ',AuthType=Basic';
+
+    if (setParametersFile) {
+        msDeployCmdArgs += ' -setParamFile=' + setParametersFile;
+    }
+    
+    if (virtualApplication) {
+        msDeployCmdArgs += ' -setParam:name=\'IIS Web Application Name\',value=\'' + webAppNameForMSDeployCmd + '/' + virtualApplication + '\'';
+    }
+    else {
+        msDeployCmdArgs += ' -setParam:name=\'IIS Web Application Name\',value=\'' + webAppNameForMSDeployCmd + '\'';
+    }
+
+    if (!removeAdditionalFilesFlag) {
+        msDeployCmdArgs += ' -enableRule:DoNotDeleteRule';
+    }
+
+    if (takeAppOfflineFlag) {
+        msDeployCmdArgs += ' -enableRule:AppOffline';
+    }
+
+    if (excludeFilesFromAppDataFlag) {
+        msDeployCmdArgs += ' -skip:Directory=App_Data';
+    }
+    
+    if (additionalArguments) {
+        msDeployCmdArgs += ' ' + additionalArguments;
+    }
+
+    var userAgent = tl.getVariable("AZURE_HTTP_USER_AGENT");
+    if (userAgent) {
+        msDeployCmdArgs += ' -userAgent:' + userAgent;
+    }
+    tl.debug(tl.loc('ConstructedmsDeploycomamndlinearguments'));
+    return msDeployCmdArgs;
+}
+
+export async function executeMSDeployCmd(msDeployCmdArgs: string, azureRMWebAppConnectionDetails, webAppUri: string) {
+    try {
+        if(webAppUri) {
+            tl.setVariable(webAppUri, azureRMWebAppConnectionDetails["destinationUrl"]);
+        }
+        var msDeployPath = await getMSDeployFullPath();
+            var statusCode = await tl.exec(msDeployPath, msDeployCmdArgs, <any> {failOnStdErr: true});
+            if ( statusCode === 0 ) {
+                tl.debug(tl.loc('WebappsuccessfullypublishedatUrl0',azureRMWebAppConnectionDetails["destinationUrl"]));                
+                var deploymentResult = await azureRmUtil.updateDeploymentStatus(azureRMWebAppConnectionDetails, true);
+                tl.debug(deploymentResult);
+            }
+            else {
+                tl.debug(tl.loc('Failedtodeploywebsite'));
+                var deploymentResult = await azureRmUtil.updateDeploymentStatus(azureRMWebAppConnectionDetails, false);
+                tl.debug(deploymentResult);
+                onError(tl.loc('Failedtodeploywebsite'));
+            }
+
+    }
+    catch(error) {
+       onError(error);
+    }
+}
+
+async function getMSDeployFullPath() {
```

Comment:
For the current sprint, the task is made to be in par with existing task.(By default, MSDeploy will be executed).